packages/web/components/place-limit-tool/index.tsx

```diff
@@ -201,55 +217,85 @@ export const PlaceLimitTool: FunctionComponent<PlaceLimitToolProps> = observer(
       );
     }, [swapState.marketState.selectableAssets, swapState.quoteAsset]);
 
-    const { outAmountLessSlippage, outFiatAmountLessSlippage } = useMemo(() => {
+    const { amountWithSlippage, fiatAmountWithSlippage } = useMemo(() => {
       // Compute ratio of 1 - slippage
       const oneMinusSlippage = new Dec(1).sub(slippageConfig.slippage.toDec());
+      const onePlusSlippage = new Dec(1).add(slippageConfig.slippage.toDec());
 
       // Compute out amount less slippage
-      const outAmountLessSlippage =
-        swapState.marketState.quote && swapState.marketState.toAsset
-          ? new IntPretty(
-              swapState.marketState.quote.amount.toDec().mul(oneMinusSlippage)
-            )
-          : undefined;
+      const amountWithSlippage =
+        quoteType === "out-given-in"
+          ? swapState.marketState.quote
+            ? new IntPretty(
+                swapState.marketState.quote.amount.toDec().mul(oneMinusSlippage)
+              )
+            : undefined
+          : new IntPretty(
+              swapState.marketState.inAmountInput?.amount
+                ?.toDec()
+                .mul(onePlusSlippage) ?? new Dec(0)
+            );
```

Comment:
Nested ternary operators make it complex to read this variable. Thoughts on using a self-calling function with if statements to improve this?
```suggestion
      const amountWithSlippage = (() => {
  if (quoteType === "out-given-in") {
    return swapState.marketState.quote
      ? new IntPretty(
          swapState.marketState.quote.amount.toDec().mul(oneMinusSlippage)
        )
      : undefined;
  } else {
    return new IntPretty(
      swapState.marketState.inAmountInput?.amount
        ?.toDec()
        .mul(onePlusSlippage) ?? new Dec(0)
    );
  }
})();
```

Reply:
I refactored this for some changes and added it to a hook in `useSwap`.